### PR TITLE
package/libs/libevent: Update to 2.1.11

### PR DESCRIPTION
**Pull request by diizzyy**
Update libevent to 2.1.11
Use CMake instead GNU Autotools
Backport following commits:
libevent/libevent@f05ba67
..and partially
libevent/libevent@7201062
to fix compilation

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>

### CODE REVIEW

**Comment by diizzyy** [comment]
Compile tested: mvebu, Linksys WRT3200ACM, OpenWrt master
Run tested: mvebu, Linksys WRT3200ACM, OpenWrt master

**Comment by diizzyy** [comment]
Compile tested: ipq40xx, Linksys EA6350v3, OpenWrt master
Run tested: ipq40xx, Linksys EA6350v3, OpenWrt master

File: package/libs/libevent2/Makefile [hunk 1]
```diff
@@ -108,22 +106,25 @@ define Package/libevent2-pthreads/description
...
 
-MAKE_FLAGS += \
-	CFLAGS="$(TARGET_CFLAGS)"
+CMAKE_OPTIONS += \
+	-DEVENT__DISABLE_BENCHMARK:BOOL=ON \
+	-DEVENT__DISABLE_DEBUG_MODE:BOOL=ON \
+	-DEVENT__DISABLE_REGRESS:BOOL=ON \
+	-DEVENT__DISABLE_SAMPLES:BOOL=ON \
+	$(if $(CONFIG_PACKAGE_libevent2-openssl),-DEVENT__DISABLE_OPENSSL:BOOL=OFF,-DEVENT__DISABLE_OPENSSL:BOOL=ON) \
+	$(if $(CONFIG_PACKAGE_libevent2-pthreads),-DEVENT__DISABLE_THREAD_SUPPORT:BOOL=OFF,-DEVENT__DISABLE_THREAD_SUPPORT:BOOL=ON) \
```
Comment (dengqf6) [inline package/libs/libevent2/Makefile]: If you use conditional configure options, add the variables to `PKG_CONFIG_DEPENDS`
Ref: openwrt/packages@41814de0fcc5

Comment (diizzyy) [reply package/libs/libevent2/Makefile]: Thanks, I'll have a look at it next week

Comment (diizzyy) [reply package/libs/libevent2/Makefile]: Hi, sorry I completely forgot about this!

I'm not entirely sure what you're suggesting here, 
```
PKG_CONFIG_DEPENDS := CONFIG_PACKAGE_libevent2-openssl \
PKG_CONFIG_PACKAGE_libevent2-mbedtls
```
?

**Comment by n8v8R** [comment]
Will this be automatically uplifted to the 19.x branch?

**Comment by diizzyy** [comment]
No idea, I have no intention of testing and/or backporting it to 19.X.

**Comment by n8v8R** [comment]
Except for testing Master there is not much use if the package only lives there until Master gets forked into the next branch, who knows when that will be - at least probably not for another year.

File: package/libs/libevent2/Makefile (hunk 1 above)
Comment (dengqf6) [reply package/libs/libevent2/Makefile]: Yes

**Comment by dengqf6** [comment]
ping @ynezz @dedeckeh

**Comment by dedeckeh** [comment]
Pulled into my staging tree; thx

**Pull request closed without merging**